src/main/java/hu/elte/eserial/annotation/EserialAnnotation.java

```diff
@@ -0,0 +1,27 @@
+package hu.elte.eserial.annotation;
+
+import java.lang.annotation.Annotation;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+import hu.elte.eserial.annotation.enumeration.EserialAnnotationType;
+import hu.elte.eserial.util.AnnotationUtils;
+
+/**
+ * The parent annotation for all annotations influencing the serialization/deserialization process.
+ * Without having this on an annotation, it will not be considered.<br>
+ */
+@Retention(RetentionPolicy.RUNTIME)
+public @interface EserialAnnotation {
+    /**
+     * Higher values indicate higher priority, meaning the annotation will be processed sooner
+     * than one with a lower priority.<br>
+     * Ordering two annotations with the same priority results in undefined behavior.<br>
```

Comment:
Can't we just fallback to the order in which they were declared?

Reply:
Some of the annotations are meant for getters/setters/fields. Others are supposed to go on class declarations. There is no relation between those, so an explicit priority is necessary.

Reply:
I see!